arrMethodsLab.js

```diff
@@ -0,0 +1,409 @@
+  if (currentElem === 0) {
+    return total + 1
+  }
+}, 0)
+
+console.log(countZeroes)
+
+// ```
+// Input: [1,0,0,2,0,3,0,0,0]
+// Output: 6
+  console.log(" ")
+//
+// 8. Write a function that doubles every element in an array.
+console.log("---------- #8 ----------")
+
+let arr9 = [1, 2, 3, 4]
+let doubled = []
+arr9.forEach((currentElem) => {
+  doubled.push(currentElem + currentElem)
+})
+
+console.log(doubled)
+
+// ```
+// Input: [1,2,3,4]
+// Output: [2,4,6,8]
+// ```
+
+console.log(" ")
+
+// 9. Write a function that finds the sum of every element in an array doubled
+console.log("---------- #9 ----------")
+
+let arr10 = [1, 2, 3, 4, 5]
+let sumOfAllDoubled = arr10.reduce((total, currentElem) => {
+  console.log(total, currentElem)
+  return total + (currentElem * 2)
+}, 0)
+
+console.log(sumOfAllDoubled)
+
+// ```
+// Input: [1,2,3,4]
+// Output: 20
+// ```
+//
+// 10. Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.  Return a new array of every element of arr that can be evenly divided by divisor.
+console.log("---------- #10 ----------")
+
+
+let arr11 = [4,5,6,7,8,9,10,11,12,13,14,15]
+let divisor = 5
+
+const elementDivisibleBy = (divisor, arr) => {
+  return arr.filter((el) => {
+    if (el % 5 === 0) {
+      return el
+    }
+  })
+}
+
+console.log(elementDivisibleBy(divisor, arr11))
+
+console.log(" ")
+
+// ```
+// Input: 5, [4,5,6,7,8,9,10,11,12,13,14,15]
+// Output: [5,10,15]
+// ```
+//
+// 11. Write a `numberTimesIdx` that uses `map` and multiples each number in the array by its index.
+
+console.log("---------- #11 ----------")
+
+let arr12 = [6, 7, 8, 9]
+
+const numberTimesIdx = (arr) => {
+  let result = arr.map((currentElem, index) => {
+    console.log(index, currentElem)
+    return index * currentElem
+  })
+  return result
+}
+
+console.log(numberTimesIdx(arr12))
+
+console.log(" ")
+// ```
+// Input: [6,7,8,9]
+// Output: [0,7,16,27]
+// ```
+//
+// 12. Write a function that returns whether or not every value in an array is a positive even number that doesn't end in 0.
+console.log("---------- #12 ----------")
+
+let arr13 = [2, 4, 6, 8, 12]
+let arr14 = [2, 4, 6, 8, 10]
+
+const posEvenCheck = (arr) => {
+  let posCheck = arr.every((currentElem) => {
+    if (currentElem > 0 && currentElem % 2 === 0 && currentElem % 10 !== 0) {
+      return true
+    } else {
+      return false
+    }
+  })
+  return posCheck
+}
+
+console.log(posEvenCheck(arr13))
+console.log(posEvenCheck(arr14))
+
+// ```
+// Input: [2,4,6,8,12]
+// Output: true
+//
+// Input: [2,4,6,8,10]
+// Output: false
+// ```
+
+console.log(" ")
+
+// 13. Write a function that returns the sum of all odd numbers in an Array
+
+console.log("---------- #13 ----------")
+
+
+let arr15 = [2, 3, 4, 5]
+let arr16 = [1, 2, 3, 4, 5]
+
+const sumOfOdd = (arr) => {
+  let oddCheck = arr.reduce((total, currentElem) => {
+    if (currentElem % 2 === 1) {
+      return currentElem
+    }
+  })
+}
+console.log(sumOfOdd(arr15))
+
```

Comment:
For this I think it would beneficial to use two different array methods. First on to filter out the odds, then the second to reduce the numbers into a sum .